### PR TITLE
Swerve cleanup and anti tip

### PR DESCRIPTION
**Pull request by witcpalek**
This needs to be tested and tuned.

### CODE REVIEW

File: src/main/cpp/Robot.cpp [hunk 1]
```diff
@@ -85,9 +85,13 @@ void Robot::RobotInit()
  */
 void Robot::RobotPeriodic()
 {
+#ifdef ENABLE_VISION
+
     static int mycounter = 0;
     static int my2ndcounter = 0;
     mycounter++;
+#endif
```
Comment (nickfrisch) [inline src/main/cpp/Robot.cpp:93]: Should the myCounter variable be removed?  Seems like deprecated debugging.  Doesn't need to be removed for this pull request, just making a note.

**Review by nickfrisch — APPROVED** [review]

File: src/main/cpp/chassis/swerve/driveStates/RobotDrive.cpp [hunk 2]
```diff
@@ -50,91 +50,106 @@ RobotDrive::RobotDrive() : ISwerveDriveState::ISwerveDriveState(),
...
+        {
+            auto adjust = m_maxspeed * m_tippingConstant * pitch.to<double>();
+            chassisMovement.chassisSpeeds.vx += adjust;
+        }
+
+        auto roll = chassis->GetRoll();
+        if (std::abs(roll.to<double>()) > m_tippingTolerance)
+        {
+            auto adjust = m_maxspeed * m_tippingConstant * roll.to<double>();
+            chassisMovement.chassisSpeeds.vy += adjust;
```
Comment (nickfrisch) [inline src/main/cpp/chassis/swerve/driveStates/RobotDrive.cpp:69]: Same question as above, should this be switched to subtraction to counter tipping?

File: src/main/cpp/Robot.cpp (hunk 1 above)
Comment (witcpalek) [reply src/main/cpp/Robot.cpp]: Yes there is a story for this

File: src/main/cpp/chassis/swerve/driveStates/RobotDrive.cpp [hunk 3]
```diff
@@ -50,91 +50,106 @@ RobotDrive::RobotDrive() : ISwerveDriveState::ISwerveDriveState(),
...
-        auto ratio = m_maxspeed.to<double>() / maxCalcSpeed;
-        m_flState.speed *= ratio;
-        m_frState.speed *= ratio;
-        m_blState.speed *= ratio;
-        m_brState.speed *= ratio;
+        auto pitch = chassis->GetPitch();
+        if (std::abs(pitch.to<double>()) > m_tippingTolerance)
+        {
+            auto adjust = m_maxspeed * m_tippingConstant * pitch.to<double>();
+            chassisMovement.chassisSpeeds.vx += adjust;
```
[the remark this answers is not in the document]
Comment (witcpalek) [reply src/main/cpp/chassis/swerve/driveStates/RobotDrive.cpp:62]: I think these are right, but need to double check when testing

File: src/main/cpp/chassis/swerve/driveStates/RobotDrive.cpp (hunk 2 above)
Comment (witcpalek) [reply src/main/cpp/chassis/swerve/driveStates/RobotDrive.cpp:69]: Need to verify when testing

**Pull request merged**